akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala

```diff
@@ -137,6 +212,31 @@ final class EventEnvelope[Event](
     val source: String,
     val tags: Set[String]) {
 
+  def this(
+      offset: Offset,
+      persistenceId: String,
+      sequenceNr: Long,
+      eventOption: Option[Event],
+      timestamp: Long,
+      eventMetadata: Option[Any],
```

Comment:
in the same spirit the metadata could be lazy, but I'm not sure how important that is